projects/common/src/time/time-range.service.ts

```diff
@@ -103,6 +101,12 @@ export class TimeRangeService {
     return this;
   }
 
+  public setDefaultTimeRange(timeRange: TimeRange): void {
+    if (!this.currentTimeRange) {
```

Comment:
This seems to work the same as `setTimeRange`. Since it is not really maintaining defaultTimeRange as a separate property, why can't we just call setTimeRange directly on the caller side if the currentTimeRange is not set?

Reply:
By using setTimeRange on the caller side for the purpose of setting the default, it allows the caller to not respect the priority that query params should be used as default if available, and otherwise use this.

The purpose of this is to make it clear that the default logic resides with the caller, but the query params still take precedence over it. This way deeplinks with query params are still respected when the caller uses this. Plus `currentTimeRange` is private

Reply:
okay

Reply:
But who is setting currentTimeRange for the first load?

Reply:
whoever calls this (nav list)